src/vm/gcenv.ee.cpp

```diff
@@ -788,10 +797,11 @@ void WalkMovedReferences(uint8_t* begin, uint8_t* end,
                                !fBGC);
 }
 
-void GCToEEInterface::DiagWalkSurvivors(void* gcContext)
+void GCToEEInterface::DiagWalkSurvivors(void* gcContext, bool fCompacting)
 {
 #if defined(GC_PROFILING) || defined(FEATURE_EVENT_TRACE)
-    if (ShouldTrackMovementForProfilerOrEtw())
+    if (ShouldTrackSurvivorsForProfilerOrEtw() ||
```

Comment:
The naming seems to mix the concepts of surviving and moving. What is supposed to happen for an object which survives a compacting GC but did not move?

I don't need for us to embark on a refactoring spree to update naming, but comments could go a long way to clarifying exactly what we expect to happen vs. whatever the current name happens to be.